.github/workflows/benchmarks_run.yml

```diff
@@ -16,15 +16,50 @@ on:
         required: false
         type: string
   pull_request:
-    types: [labeled]
+    # Add the `labeled` type to the default list.
+    types: [labeled, opened, synchronize, reopened]
 
 jobs:
+  pre-checks:
+    runs-on: ubuntu-latest
+    if: github.repository == 'SciTools/iris'
+    outputs:
+      overnight: ${{ steps.overnight.outputs.overnight }}
+      branch: ${{ steps.branch.outputs.branch }}
+      continue: ${{ steps.check-continue.outputs.continue }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+      - id: locks-changed
+        uses: marceloprado/has-changed-path@df1b7a3161b8fb9fd8c90403c66a9e66dfde50cb
+        with:
+          paths: requirements/locks/*.lock setup.py pyproject.toml
+      - id: overnight
+        if: github.event_name != 'pull_request'
+        run: echo "overnight=true" >> "$GITHUB_OUTPUT"
+      - id: branch
+        if: >
+          github.event_name == 'pull_request'
+          &&
+          (
+            steps.locks-changed.outputs.changed == 'true'
+            ||
+            github.event.label.name == 'benchmark_this'
+          )
+        run: echo "branch=true" >> "$GITHUB_OUTPUT"
+      - id: check-continue
+        if: >
+          steps.overnight.outputs.overnight == 'true'
+          ||
+          steps.branch.outputs.branch == 'true'
+        run: echo "continue=true" >> "$GITHUB_OUTPUT"
+
+
   benchmark:
-    if: >
-      github.repository == 'SciTools/iris' && 
-      (github.event_name != 'pull_request' || 
-      github.event.label.name == 'benchmark_this')
     runs-on: ubuntu-latest
+    needs: pre-checks
+    if: needs.pre-checks.outputs.continue == 'true'
```

Comment:
This would make more sense to me, if it was just checking if either of needs.pre-checks.outputs.branch or needs.pre-checks.outputs.overnight is equal to 'true'.

Since this is the only place continue is referenced, you can remove it in all places.